### PR TITLE
Feature/fix pdf generation race condition

### PR DESCRIPTION
**Pull request by wontknow**
## Zusammenfassung
(Was wurde gemacht? Warum?)

## Verwandte Issues
Closes #123

## Änderungen
- ...
- ...

## Wie testen?
1. Backend starten: ...
2. Frontend: ...
3. Erwartet: ...

## Screenshots / Logs
(bei UI)

## Checkliste (DoD)
- [ ] Funktionalität erfüllt (Akzeptanzkriterien)
- [ ] Readme/Docs aktualisiert (falls nötig)
- [ ] Keine Secrets/Keys committed
- [ ] Lint/Build grün
- [ ] Manuelle Tests (happy path & edge cases)
- [ ] CORS geprüft (Dev/Prod)  
- [ ] PDF erzeugt & visuell geprüft  
- [ ] (optional) E-Rechnung Validierung dokumentiert

### CODE REVIEW

File: backend/routers/summary_invoices.py [hunk 1]
```diff
@@ -86,6 +88,45 @@ def create_summary(
...
+            bg_session.close()
+
+    pdf_thread = threading.Thread(
+        target=generate_pdf_background,
+        daemon=True,
+        name=f"PDF-Summary-{summary_invoice.id}",
+    )
+    logger.debug(f"🖨️ Thread created: {pdf_thread.name}")
+    pdf_thread.start()
+    logger.debug(f"🖨️ Thread started")
```
[the remark this answers is not in the document]
Comment (wontknow) [reply backend/routers/summary_invoices.py]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/wontknow/Billino/pull/257#discussion_r2647313171)

Make the helper class, implement the class usage in both usecases and wirte meaningful tests for it. Check the old tests, if all tests run correctly. Then use black and isort in the venv to check if everything works correctly

File: backend/services/pdf_generation_service.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+            f"✅ PDF generated for invoice {invoice_id} (PDF ID: {stored_pdf.id}, size: {len(pdf_base64)} bytes)"
+        )
+        return True
+
+    except Exception as e:
+        logger.error(
+            f"❌ Failed to generate PDF for invoice {invoice_id}: {str(e)}",
+            exc_info=True,
+        )
+        return False
```
[the remark this answers is not in the document]
Comment (wontknow) [reply backend/services/pdf_generation_service.py]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/wontknow/Billino/pull/257#discussion_r2647313167)

use npm run format and lint after the fix

File: backend/services/pdf_generation_service.py [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+
+        # Check if PDF already exists
+        existing_pdf = session.exec(
+            select(StoredPDF).where(StoredPDF.summary_invoice_id == summary_invoice_id)
+        ).first()
+        if existing_pdf:
+            logger.debug(
+                f"ℹ️ PDF for summary invoice {summary_invoice_id} already exists (ID: {existing_pdf.id})"
+            )
+            return False
```
[the remark this answers is not in the document]
Comment (wontknow) [reply backend/services/pdf_generation_service.py]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/wontknow/Billino/pull/257#discussion_r2648589995)

On Conflict do nothing

After implementing the strategy without breaking any functionality and existing code, make tests, run existing tests, use format and lint AND check potential issues by your code. Review it twice for SOLID and Clean Code and FUnctionality and best practices before giving it to me as review!

File: backend/models/stored_pdf.py [hunk 4]
```diff
@@ -17,9 +17,28 @@ class StoredPDFBase(SQLModel):
 
 
 class StoredPDF(StoredPDFBase, table=True):
-    """Database model for stored PDFs"""
+    """Database model for stored PDFs with unique constraints to prevent duplicates
```
[the remark this answers is not in the document]
Comment (wontknow) [reply backend/models/stored_pdf.py]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/wontknow/Billino/pull/257#r2648623056)

File: backend/services/background_pdf_generator.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+                    f"❌ [THREAD] Background PDF generation failed for {entity_type} {entity_id}: {str(e)}",
+                    exc_info=True,
+                )
+            finally:
+                bg_session.close()
+
+        # Create and start the daemon thread
+        pdf_thread = threading.Thread(
+            target=generate_pdf_background,
+            daemon=True,
```
[the remark this answers is not in the document]
Comment (wontknow) [reply backend/services/background_pdf_generator.py]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/wontknow/Billino/pull/257#r2648601238)

File: backend/services/pdf_generation_service.py [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+        # if we already know one exists. The unique constraint on invoice_id provides
+        # the definitive protection against duplicates during concurrent requests.
+        existing_pdf = session.exec(
+            select(StoredPDF).where(StoredPDF.invoice_id == invoice_id)
+        ).first()
+        if existing_pdf:
+            logger.debug(
+                f"ℹ️ PDF for invoice {invoice_id} already exists (ID: {existing_pdf.id})"
+            )
+            return False
```
[the remark this answers is not in the document]
Comment (wontknow) [reply backend/services/pdf_generation_service.py]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/wontknow/Billino/pull/257#r2648623063)

File: backend/services/pdf_generation_service.py [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+    except Exception as e:
+        logger.error(
+            f"❌ Failed to generate PDF for invoice {invoice_id}: {str(e)}",
+            exc_info=True,
+        )
+        return False
+
+
+def generate_pdf_for_summary_invoice(
+    session: Session, summary_invoice_id: int, recipient_name: str = None
```
[the remark this answers is not in the document]
Comment (wontknow) [reply backend/services/pdf_generation_service.py]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/wontknow/Billino/pull/257#r2647313170)

File: frontend/src/services/pdfs.ts [hunk 8]
```diff
@@ -42,6 +42,62 @@ export class PDFsService {
...
+      const pdfResponse: StoredPDF = await ApiClient.get<StoredPDF>(
+        `/pdfs/by-invoice/${invoiceId}`
+      );
+      log.debug(`✅ PDF Loaded: Invoice ${invoiceId} PDF found`);
+      return PDFsService.convertBase64ToPdfBlob(pdfResponse);
+    } catch (error: unknown) {
+      if (error instanceof ApiError) {
+        if (error.status === 404) {
+          log.warn(`⚠️ PDF Not Found: Invoice ${invoiceId} - Triggering generation...`);
+          return PDFsService.createPdfForInvoice(invoiceId);
```
[the remark this answers is not in the document]
Comment (wontknow) [reply frontend/src/services/pdfs.ts]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/wontknow/Billino/pull/257#r2648601259)

File: frontend/src/services/pdfs.ts [hunk 9]
```diff
@@ -42,6 +42,62 @@ export class PDFsService {
...
+        }
+        log.error(`❌ PDF Loading Error [${error.status}]: Invoice ${invoiceId}`, error.detail);
+      }
+      throw error;
+    }
+  }
```
[the remark this answers is not in the document]
Comment (wontknow) [reply frontend/src/services/pdfs.ts]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/wontknow/Billino/pull/257#r2648590004)

File: frontend/src/services/pdfs.ts [hunk 10]
```diff
@@ -42,6 +42,62 @@ export class PDFsService {
...
+          return PDFsService.createPdfForSummaryInvoice(summaryInvoiceId, recipientName);
+        }
+        log.error(
+          `❌ PDF Loading Error [${error.status}]: Summary invoice ${summaryInvoiceId}`,
+          error.detail
+        );
+      }
+      throw error;
+    }
+  }
```
[the remark this answers is not in the document]
Comment (wontknow) [reply frontend/src/services/pdfs.ts]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/wontknow/Billino/pull/257#r2647313175)

**Comment by wontknow** [comment]
@copilot open a new pull request to apply changes based on the comments in [this thread](https://github.com/wontknow/Billino/pull/257#pullrequestreview-3614309936)

**Comment by wontknow** [comment]
@copilot open a new pull request to apply changes based on the comments in [this thread](https://github.com/wontknow/Billino/pull/257#pullrequestreview-3614311403)
